add requirements_dev.txt file (#198)
illinois-ceesd/mirgecom: requirements_dev.txt

# Useful development packages.
# Install them with 'pip install -r requirements_dev.txt'.

memory-profiler
jedi
pudb
pytest-pudb
pyprof2calltree
pyinstrument
# Might fail on some machines, you might have to install from git, with the commented line below:
vmprof; platform_machine == "x86_64" 

> # git+https://github.com/vmprof/vmprof-python.git

Reviewer: Just occurred to me: We may want to install that instead of the version from the package index, as that has `vmprofshow flat`, whereas the pypi version doesn't.

Author: Done

# Useful development packages.
# Install them with 'pip install -r requirements_dev.txt'.

memory-profiler

> pudb

Reviewer: ```suggestion
jedi
pudb
```

> pudb

Reviewer: ```suggestion
pudb
pytest-pudb
pyprof2calltree
```

pyinstrument

> vmprof; platform_machine == "x86_64"

Reviewer: FWIW, `pip install vmprof` fails for me, I have to install their repo. Maybe mention that?

Author: Weird, it works for me. I'll add a note.

> vmprof; platform_machine == "x86_64" # Might fail on some machines, you might have to install the package from the repo.

Reviewer: ```suggestion
# Might fail on some machines, you might have to install from git, with the commented line below:
vmprof; platform_machine == "x86_64" 
# git+https://github.com/vmprof/vmprof-python.git
```
